src/compositions/Timeline/Timeline.tsx

```diff
@@ -1,42 +1,220 @@
-import { CSSProperties, FC, ReactElement } from 'react'
+import React, { CSSProperties, FC, ReactElement, useEffect, useRef, useState } from 'react'
+import {
+    Timeline as ReactTimeline,
+    TimelineAction,
+    TimelineEffect,
+    TimelineRow,
+    TimelineState,
+} from '@xzdarcy/react-timeline-editor'
+
+import { Icon, TimelineCaptionClip, TimelineVideoClip } from '@app-components'
 
 import './Timeline.scss'
 
 export interface TimelineProps {
     cues: VTTCue[]
     activeCueIndex?: number
-    onCueClick: (index: number) => void
+    videoLength: number
     currentTime: number
+    isPlaying: boolean
+    onCueClick: (event: any, params: {
+        action: TimelineAction,
+        row: TimelineRow,
+        time: number
+    }) => void
+    onCueMove: (params: { action: TimelineAction, row: TimelineRow, start: number, end: number }) => boolean
+    onSetCurrentTime: (time: number) => void
     className?: string
 }
 
 export const Timeline: FC<TimelineProps> = ({
     cues,
     activeCueIndex,
+    videoLength,
     currentTime,
+    isPlaying,
     onCueClick,
+    onCueMove,
+    onSetCurrentTime,
     className = '',
-}): ReactElement => (
-    <div
-        className={`timeline ${className}`}
-        style={{ '--current-time': currentTime } as CSSProperties}
-    >
-
-        <div className="timeline__timings"></div>
-
-        <ul className="timeline__cue-list">
-            {cues.map((cue, index) => (
-                <li key={index}
-                    className={`timeline__cue-item ${activeCueIndex === index ? 'timeline__cue-item--active' : ''}`}
-                    style={{ '--cue-start-time': cue.startTime, '--cue-end-time': cue.endTime } as CSSProperties}
+}): ReactElement => {
+    const [scale, setScale] = useState(5)
+    const [scaleWidth, setScaleWidth] = useState(50)
+    const [timelineData, setTimelineData] = useState<TimelineRow[]>([])
+
+    const timelineState = useRef<TimelineState>(null)
+
+    const timelineEffects: Record<string, TimelineEffect> = {
+        caption: {
+            id: 'caption',
+            name: 'caption',
+        },
+        video: {
+            id: 'video',
+            name: 'video',
+        },
+    }
+
+    const increaseZoom = () => {
+        if (scale > 2) {
+            setScale(scale / 2)
+        } else {
+            setScaleWidth(scaleWidth * 2)
+        }
+    }
+
+    const decreaseZoom = () => {
+        if (timelineState.current && videoLength * (scaleWidth / Math.floor(scale)) < timelineState.current.target.clientWidth) {
+            return
+        }
+
+        if (scaleWidth > 50) {
```

Comment:
Where is this `50` based on? Perhaps it would be good to extract this to a constant with a descriptive name